steward/src/cork.rs

```diff
@@ -94,6 +105,140 @@ impl steward::contract_call_server::ContractCall for CorkHandler {
     }
 }
 
+pub struct DirectCorkHandler;
+
+#[async_trait]
+impl steward::contract_call_server::ContractCall for DirectCorkHandler {
+    async fn submit(
+        &self,
+        request: Request<SubmitRequest>,
+    ) -> Result<Response<SubmitResponse>, Status> {
+        debug!("received contract call request");
+        let request = request.get_ref();
+
+        let contract_call_data = match request.call_data.clone() {
+            Some(call) => call,
+            None => return Err(tonic::Status::invalid_argument("err")),
+        };
+
+        let config = APP.config();
+
+        let keystore = path::Path::new(&config.keystore);
+        let keystore = FsKeyStore::create_or_open(keystore).expect("Could not open keystore");
+
+        let name = &config
+            .keys
+            .delegate_key
+            .parse()
+            .expect("Could not parse name");
+        let key = keystore.load(name).expect("Could not load key");
+
+        let key = key
+            .to_pem()
+            .parse::<k256::elliptic_curve::SecretKey<k256::Secp256k1>>()
+            .expect("Could not parse key");
+
+        let wallet: LocalWallet = Wallet::from(key);
+
+        let eth_host = config.ethereum.rpc.clone();
+        let provider = Provider::<Http>::try_from(eth_host.clone())
+            .unwrap_or_else(|_| panic!("could not get provider from eth_host {}", eth_host));
+        let chain_id = provider
+            .get_chainid()
+            .await
+            .expect("Could not retrieve chain ID");
+
+        let chain_id =
+            downcast_to_u64(chain_id).expect("Chain ID overflowed when downcasting to u64");
+        let client = SignerMiddleware::new(provider, wallet.clone().with_chain_id(chain_id));
+
+        let cellar_address = request.cellar_id.clone();
+
+        if let Err(err) = cellars::validate_cellar_id(&cellar_address) {
+            return Err(tonic::Status::invalid_argument(err));
+        }
+
+        let address: Address = cellar_address
+            .parse()
+            .expect("could not parse cellar address");
+
+        match contract_call_data {
+            AaveV2Stablecoin(call) => {
+                let contract = AaveV2StablecoinCellar::new(address, Arc::new(client));
+
+                match call.function.unwrap() {
+                    Function::EnterStrategy(_) => contract_call(contract.enter_strategy()).await,
```

Comment:
Since these calls can fail, once `contract_call` is updated, the case where `await` returns an error result needs to be bubbled back up and returned as a `tonic` error to the caller.